### ISSUE TITLE
no-unsafe-any and {} | null | undefined

### ISSUE THREAD

**Issue by stephanedr** [palantir/tslint#4816]
### Bug Report

- __TSLint version__: 5.18.0
- __TypeScript version__: 3.5.3
- __Running TSLint via__: VSCode

#### Reproduction using [TSLint Playground](https://palantir.github.io/tslint-playground)

https://palantir.github.io/tslint-playground/?saved=N4Igxg9gJgpiBcJZgDYEMBOMAEAzArgHZgAuAlhIXgBQTzbAC+2APtofiiq9kbLmUIwoASnoA3CGSgBuADqFc1JgG05IXBAjqAuiPmEQAGnCUBAcwQhgC7HezqMnGAGd19G1Xvf1hCAFoiFzRcGH80QgBPd2wSJxhbb0YFZMNGIA

Note that the issue in not raised (whereas it is under VSCode). Maybe due to different TSC options?

#### TypeScript code being linted

```ts
declare function f(o: {} | null | undefined): void;
f({}["foo"]); // Unsafe use of expression of type 'any'. (no-unsafe-any)
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-unsafe-any": true
     }
}
```

#### Actual behavior
Error reported in the comment.

#### Expected behavior
No error.
The spec says: "Uses are only allowed if they would work for `{} | null | undefined`". So I understand this should be OK.
Additionally I consider this is safe.

**Comment by JoshuaKGoldberg** [comment]
Following up on this @stephanedr - your issue is with `{}["foo"]`. In this code, `{}` is inferred to be the "empty object" type. That means it's an object with no properties. There's no `["foo"]` property on it.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
